docs/default.nix

```
@@ -192,27 +201,48 @@ let
                   else
                     maintainer.name;
 
-                joinItems =
-                  items:
-                  if builtins.length items <= 2 then
-                    builtins.concatStringsSep " and " items
-                  else
-                    builtins.concatStringsSep ", " (
-                      lib.dropEnd 1 items ++ [ "and ${lib.last items}" ]
-                    );
-
                 renderedMaintainers = joinItems (map renderMaintainer maintainers);
 
                 maintainersText =
                   if maintainers == [ ] then
-                    "This module has no [dedicated maintainers](../../modules.md#maintainers)."
+                    ''
+                      This module has no [dedicated maintainers](../../modules.md#maintainers).
+                    ''
+                  else
+                    ''
+                      This module is maintained by ${renderedMaintainers}.
+                    '';
+
+                # Render homepages as hyperlinks in readme
+                #
+                # Takes the `homepages` attribute set from `meta.nix` and
+                # combines it with homepages extracted from  the `packages`
+                # list, also from `meta.nix`.
+                homepages = metadata.${module}.homepages or { };
+
+                renderHomepage = name: url: "[${name}](${url})";
+
+                renderedHomepages = joinItems (lib.mapAttrsToList renderHomepage homepages);
+
+                homepageText =
+                  if homepages == { } then
+                    ''
+                      Homepages: none listed
+                    ''
+                  else if builtins.length (builtins.attrNames homepages) == 1 then
+                    renderHomepage ''
+                      Homepage
+                    '' (builtins.head (builtins.attrValues homepages))
```

consider having homepage be either str or attrset of str:
```suggestion
                  else if builtins.isString homepages then
                    renderHomepage "Homepage" homepages
```

---

I think that this would be less intuitive as it feels odd to have `homepages` be set to one single thing rather than being set to a set that just happens to only contain one item.

---

maybe have either `homepage` or `homepages` and throw if they are both set

---

To me, that sounds like unnecessary complexity.

---

imo having to set an attr name that is ignored is worse

---

Alright; `homepages` now takes either str or set of str.